functions/go/fix/fixpkg/fix.go

```diff
@@ -230,15 +230,18 @@ func (s *Fix) FunctionsInPkg(nodes []*yaml.RNode, pkgPath string) []v1.Function
 		}
 		fnSpec := runtimeutil.GetFunctionSpec(node)
 		if fnSpec != nil {
+			// in order to make sure there is only one function config per file,
+			// rename the file with the name(and hash(name)) of the function config resource
+			fnFileName := fmt.Sprintf("%s.yaml", node.GetName())
 			// in v1, fn-config must be present in the package directory
 			// so configPath must be just the file name
-			fnFileName := filepath.Base(meta.Annotations[kioutil.PathAnnotation])
+			fnFilePath := filepath.Join(pkgPath, fnFileName)
 			res = append(res, v1.Function{
 				Image:      fnSpec.Container.Image,
 				ConfigPath: fnFileName,
 			})
 			// move the fn-config to the top level directory of the package
-			meta.Annotations[kioutil.PathAnnotation] = filepath.Join(pkgPath, fnFileName)
+			meta.Annotations[kioutil.PathAnnotation] = fnFilePath
```

Comment:
How do we handle path conflicts ?

Reply:
Actually, conflicts happen if there are 2 function configs with same name. I think it is highly unlikely. We need not optimize for that. So to keep it simple, I just used resource name.